cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedTable.java

```diff
@@ -120,9 +123,27 @@ public IndexedTable(String name, Table table, Table index, byte[][] columnsToInd
     this.table = table;
     this.index = index;
     this.indexedColumns = Sets.newTreeSet(Bytes.BYTES_COMPARATOR);
-    for (byte[] col : columnsToIndex) {
-      this.indexedColumns.add(col);
+    this.hasColumnWithZero = hasZeros(columnsToIndex);
+    Collections.addAll(this.indexedColumns, columnsToIndex);
+  }
+
+  /**
+   * Checks if a set of columns contains the DELIMITER_BYTE.
+   * This is needed because only when a column has a zero value in it do we need to check for false positive in the scan
+   * on the index by checking against the column's value in the data table
```

Comment:
It would also be useful to explain why ]0 in an indexColumn can cause false positives. For example by giving an example of a=b\0c looks the same as a\0b=c. It is not obvious from the code that this would be a false positive...